### PR TITLE
Cleanup existing double quote handling

### PR DESCRIPTION
**Pull request by kzhuravl**

### CODE REVIEW

**Comment by kzhuravl** [comment]
This essentially only introduces double quotes to windows, but not for linux

**Comment by yxsamliu** [comment]
Is this because Linux has the slightest chance to see paths with space, so we do not want to pollute the command line with quotes?

**Comment by kzhuravl** [comment]
> Is this because Linux has the slightest chance to see paths with space, so we do not want to pollute the command line with quotes?

I ran into some issues with the original implementation when building rocTHRUST on linux, and this patch seemed to fix it. Unfortunately, I am not familiar with whitespaces on linux. Evgeny mentioned this approach in one of the cherry-pick reviews, so I went ahead and did it.

@emankov, any thoughts on whitespaces on linux? based on my googling, whitespaces in paths happen on linux and are also handled with double or single quotes around the path... https://stackoverflow.com/questions/12902227/how-to-input-a-path-with-a-white-space

File: bin/hipcc.pl [hunk 1]
```diff
@@ -131,6 +132,14 @@ BEGIN
 $HIP_VERSION    =   $hipvars::HIP_VERSION;
 $HIP_ROCCLR_HOME =   $hipvars::HIP_ROCCLR_HOME;
 
+sub get_normalized_path {
+    if ($isWindows) {
```
Comment (emankov) [inline bin/hipcc.pl:136]: I think we can normalize paths always and unconditionally. Clang driver, for instance, does it the same way.

File: bin/hipvars.pm [hunk 2]
```diff
@@ -124,7 +128,7 @@ if (defined $HIP_RUNTIME and $HIP_RUNTIME eq "rocclr" and !defined $HIP_ROCCLR_H
 }
 
 if (not defined $HIP_PLATFORM) {
-    if (can_run("\"$HIP_CLANG_PATH/clang++\"") or can_run("clang++")) {
+    if (can_run($doubleQuote . "$HIP_CLANG_PATH/clang++" . $doubleQuote) or can_run("clang++")) {
         $HIP_PLATFORM = "amd";
     } elsif (can_run("$CUDA_PATH/bin/nvcc") or can_run("nvcc")) {
```
Comment (emankov) [inline bin/hipvars.pm:133]: We need CUDA_PATH double-quoted too.

File: bin/hipvars.pm [hunk 3]
```diff
@@ -61,6 +61,10 @@ sub can_run {
 }
 
 $isWindows =  ($^O eq 'MSWin32' or $^O eq 'msys');
+$doubleQuote = "";
+if ($isWindows) {
+  $doubleQuote = "\"";
```
Comment (emankov) [inline bin/hipvars.pm:66]: Let's try to leave it as an unconditional const `"\""`.

**Review by emankov — CHANGES_REQUESTED** [review]

**Comment by yxsamliu** [comment]
The failure on rocThrust could be due to some corner case that is not handled when quoting the options.

I am OK with this patch since we want to deprecate hipcc and ask people to use clang directly. I feel not worth the effort to spend too much time on hipcc.

**Comment by emankov** [comment]
> I am OK with this patch since we want to deprecate hipcc and ask people to use clang directly. I feel not worth the effort to spend too much time on hipcc.

Unfortunately, we need to fix the nvcc path in hipcc after all, cause hipcc is still going to be shipped in at least two HIP SDK releases for Windows.

**Review by emankov — APPROVED** [review]
LGTM, thanks!

**Comment by kzhuravl** [comment]
fyi, I am getting rocThrust build failures in internal ci when returning a double quote for linux path:

2023-07-22T00:57:36.221388549Z clang: error: invalid target ID 'gfx90a\:xnack+'; format is a processor name followed by an optional colon-delimited list of features followed by an enable/disable sign (e.g., 'gfx908:sramecc+:xnack-')

**Comment by kzhuravl** [comment]
Is everyone ok with not putting a double quotes around paths on linux for now? @emankov?

**Comment by yxsamliu** [comment]
> fyi, I am getting rocThrust build failures in internal ci when returning a double quote for linux path:
> 
> 2023-07-22T00:57:36.221388549Z clang: error: invalid target ID 'gfx90a:xnack+'; format is a processor name followed by an optional colon-delimited list of features followed by an enable/disable sign (e.g., 'gfx908:sramecc+:xnack-')

That's a valid target ID. I suspect it was quoted twice, e.g. "--offload-arch='gfx90a:xnack+'", then clang gets an option value containing a single quote.

**Comment by emankov** [comment]
> fyi, I am getting rocThrust build failures in internal ci when returning a double quote for linux path:
> 
> 2023-07-22T00:57:36.221388549Z clang: error: invalid target ID 'gfx90a:xnack+'; format is a processor name followed by an optional colon-delimited list of features followed by an enable/disable sign (e.g., 'gfx908:sramecc+:xnack-')

I do not see any double quotes in the message. Could you please provide us with the clang command line, if possible?

File: bin/hipcc.pl [hunk 4]
```diff
@@ -466,7 +475,7 @@ BEGIN
     if (not $isWindows and $escapeArg) {
         $arg =~ s/[^-a-zA-Z0-9_=+,.\/]/\\$&/g;
     }
-    $toolArgs .= " \"$arg\"" unless $swallowArg;
```
Comment (emankov) [inline bin/hipcc.pl:469]: Here in the first commit, the non-path, actually all the args for passing to a tool, were touched. On Linux, double quotes were removed, and on Windows, they were left. I missed it.

**Review by emankov — CHANGES_REQUESTED** [review]
The change should revise only path-containing options and leave untouched the rest of the options.

**Review by david-salinas — APPROVED** [review]
LGTM ... but, I'd like to see if we can get a Windows build to verify no regressions.

File: bin/hipcc.pl (hunk 4 above)
Comment (kzhuravl) [reply bin/hipcc.pl:469]: I had to undo this due to issues while building rocThrust.

**Review by emankov — APPROVED** [review]
LGTM, thanks!

**Pull request merged**